### PR TITLE
feat(template) : Added automation tool as makim

### LINKED ISSUE #91: Add makim as an option for automation tool
currently, the template offers `make` automatically without asking.
it would be nice to have an option for automation_tools. where it will allow makim (default) or make

### PR DESCRIPTION
**Pull request by ayeankit**
## Pull Request description

Fixes #91 

Added Makim tool as makim. 
1. added makim as dependency in pyproject.toml(template)
2. added .makim.yaml file
3. changed base.sh file
4. changed ci file(main.yaml) changed

## How to test these changes

* ```...```

## Pull Request checklists

This PR is a:
- [ ] bug-fix
- [x] new feature
- [ ] maintenance

About this PR:
- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:
- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more complexity.
- [ ] New and old tests passed locally.

## Reviewer's checklist

Copy and paste this template for your review's note:

```
## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved .
```

### CODE REVIEW

File: src/scicookie/{{cookiecutter.project_slug}}/build-system/flit-pyproject.toml [hunk 1]
```diff
@@ -120,6 +120,11 @@ dependencies = [
     "jupyter-book >= 0.15.1",
     "myst-parser >= 0.18.1",
 {% endif %}
+{%- if cookiecutter.automation_tools == "make" %}
+    "make >=0.1.6",
```
Comment (xmnlab) [inline src/scicookie/{{cookiecutter.project_slug}}/build-system/flit-pyproject.toml:124]: `make` (the correct one) is not available on pypi .. this one in pypi is a different project
for  `make`, just add it to the conda env file 
you just need to check here if the automation tool is makim, and add it if it is selected.

the same applies to all the other build system files

File: tests/smoke/base.sh [hunk 2]
```diff
@@ -70,9 +70,12 @@ $COMMAND_PREFIX pre-commit install
 
 $COMMAND_PREFIX pre-commit run --all-files --verbose
 
-$COMMAND_PREFIX make docs-build
+$COMMAND_PREFIX make docs-build
```
Comment (xmnlab) [inline tests/smoke/base.sh:73]: for all the places that uses `make` in this file  ... you will need to change them .. and add an `if` that will check if it uses make or makim and call the correct command.

remember that the `make target's name` here use `-` but `makim` uses `.`.

**Review by xmnlab — CHANGES_REQUESTED** [review]
awesome! almost there!!

some additional comments:

* also please check if there is any information about makim in the guide.md, otherwise create it
* if there is any command using `make` in the template contributing guide and template CI workflows file, please add the proper changes that will use make or makim according to the user selection.

File: tests/smoke/base.sh [hunk 3]
```diff
@@ -70,8 +70,16 @@ $COMMAND_PREFIX pre-commit install
 
 $COMMAND_PREFIX pre-commit run --all-files --verbose
 
-$COMMAND_PREFIX make docs-build
-make build
+if "{{cookiecutter.automation_tools}}" == "makim" ; then
```
Comment (xmnlab) [inline tests/smoke/base.sh:73]: sorry just remembered that the cookiecutter variables are not accessable here
so you need to use something like:

```suggestion
if command -v makim &> /dev/null; then
```

File: tests/smoke/base.sh [hunk 4]
```diff
@@ -70,8 +70,16 @@ $COMMAND_PREFIX pre-commit install
...
+  $COMMAND_PREFIX makim docs.build
+  makim build
+else
+  $COMMAND_PREFIX make docs-build 
+  make build 
+
```
Comment (xmnlab) [inline tests/smoke/base.sh:79]:
```suggestion
```

File: docs/guide.md [hunk 5]
```diff
@@ -767,6 +767,29 @@ efficient platform for running Python projects.
...
+
+## Automation Tools
+
+An automation tool is software or a platform designed to automate repetitive 
+tasks, processes, or workflows that are traditionally performed manually.
+These tools are used to streamline and optimize various operations, reduce 
+human intervention, increase efficiency, and minimize errors.
+Currently OSL is using `make` and `Makim`
+
+- [Makim](https://osl-incubator.github.io/makim):MakIm is an innovative tool
```
Comment (xmnlab) [inline docs/guide.md:779]:
```suggestion
- [Makim](https://osl-incubator.github.io/makim): MakIm is an innovative tool
```

File: docs/guide.md [hunk 6]
```diff
@@ -767,6 +767,29 @@ efficient platform for running Python projects.
...
+inspired by Make, designed to simplify target and dependency definition
+through YAML format. It introduces advanced features such as conditionals,
+arguments, and dependencies with targeted parameters. It also facilitates
+organized grouping of targets and supports user-defined variables and environment
+variables. MakIm empowers users to streamline documentation and parameterize
+targets effectively. This free and open-source tool offers improved target
+management while maintaining compatibility with familiar YAML syntax.
+
+- [Make](https://pypi.org/project/make/):Make is a versatile build automation
```
Comment (xmnlab) [inline docs/guide.md:788]:
```suggestion
- [Make](https://en.wikipedia.org/wiki/Make_(software)): Make is a versatile build automation
```

File: src/scicookie/{{cookiecutter.project_slug}}/.github/workflows/main.yaml [hunk 7]
```diff
@@ -31,9 +31,17 @@ jobs:
       run: poetry install
 
     - name: Run tests
-      run: make test
-
+      run: |
+        {% if cookiecutter.automation_tools == "makim" %}
+          makim test
```
Comment (xmnlab) [inline src/scicookie/{{cookiecutter.project_slug}}/.github/workflows/main.yaml:36]:
```suggestion
          makim tests.unittest
```

File: src/scicookie/{{cookiecutter.project_slug}}/.github/workflows/main.yaml [hunk 8]
```diff
@@ -31,9 +31,17 @@ jobs:
...
+        {% else %}
+          make test
+        {% endif %}
     - name: Run style checks
       run: |
         pre-commit install
-        make lint
+        {% if cookiecutter.automation_tools == "makim" %}
+          makim lint
```
Comment (xmnlab) [inline src/scicookie/{{cookiecutter.project_slug}}/.github/workflows/main.yaml:44]:
```suggestion
          makim tests.lint
```

**Review by xmnlab — CHANGES_REQUESTED** [review]
@ayeankit thanks for applying the suggestions pointed previously
I just added the latest suggestion, and it would be ready to go.
thank you so much

**Comment by ayeankit** [comment]
hey @xmnlab , All test cases passed. Can you review it once again and merge it? Thank you!!

File: docs/guide.md [hunk 9]
```diff
@@ -767,6 +767,29 @@ efficient platform for running Python projects.
...
 by selecting the option `None` (this is the default option).
 
+
+## Automation Tools
+
+An automation tool is software or a platform designed to automate repetitive 
+tasks, processes, or workflows that are traditionally performed manually.
+These tools are used to streamline and optimize various operations, reduce 
+human intervention, increase efficiency, and minimize errors.
+Currently OSL is using `make` and `Makim`
```
Comment (Anavelyz) [inline docs/guide.md:777]: Improve it, Here it should be SciCookie.

Comment (xmnlab) [reply docs/guide.md:777]: @ayeankit, could you open a follow-up PR to change OSL to SciCokkie, as pointed by @anavelyz, please?

thanks @anavelyz for catching that!

Comment (ayeankit) [reply docs/guide.md:777]: sure !!

**Pull request merged**